src/org/thoughtcrime/securesms/PinnedMessagesListActivity.java

```diff
@@ -0,0 +1,156 @@
+    private final DynamicLanguage dynamicLanguage = new DynamicLanguage();
+    private ConversationFragment fragment;
+
+    @Override
+    protected void onPreCreate() {
+        dynamicTheme.onCreate(this);
+        dynamicLanguage.onCreate(this);
+    }
+
+    @Override
+    public void onCreate(Bundle state, @NonNull MasterSecret masterSecret) {
+        Log.w(TAG, "onCreate()");
+
+        supportRequestWindowFeature(WindowCompat.FEATURE_ACTION_BAR_OVERLAY);
+        setContentView(R.layout.pinned_messages_activity);
+
+        TypedArray typedArray = obtainStyledAttributes(new int[]{R.attr.conversation_background});
+        int color = typedArray.getColor(0, Color.WHITE);
+        typedArray.recycle();
+
+        getWindow().getDecorView().setBackgroundColor(color);
+
+        ConversationFragment conversationFragment = new ConversationFragment();
+        conversationFragment.onlyPinned = true;
+
+        fragment = initFragment(R.id.fragment_content, conversationFragment,
+                masterSecret, dynamicLanguage.getCurrentLocale());
+
+        initializeActionBar();
+        initializeViews();
+        initializeResources();
+    }
+
+    protected void initializeActionBar() {
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar == null) throw new AssertionError();
+
+        supportActionBar.setDisplayHomeAsUpEnabled(false);
+        supportActionBar.setCustomView(R.layout.conversation_title_view);
+        supportActionBar.setDisplayShowCustomEnabled(true);
+        supportActionBar.setDisplayShowTitleEnabled(false);
+    }
+
+
+    private void initializeViews() {
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar == null) throw new AssertionError();
+
+        titleView = (ConversationTitleView) supportActionBar.getCustomView();
+        titleView.setOnBackClickedListener(view -> super.onBackPressed());
+    }
+
+    private void initializeResources() {
+        if (recipient != null) recipient.removeListener(this);
+
+        recipient = Recipient.from(this, getIntent().getParcelableExtra(ADDRESS_EXTRA), true);
+        threadId = getIntent().getLongExtra(THREAD_ID_EXTRA, -1);
+        archived = getIntent().getBooleanExtra(IS_ARCHIVED_EXTRA, false);
+        distributionType = getIntent().getIntExtra(DISTRIBUTION_TYPE_EXTRA, ThreadDatabase.DistributionTypes.DEFAULT);
+        GlideRequests glideRequests = GlideApp.with(this);
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+            LinearLayout conversationContainer = ViewUtil.findById(this, R.id.conversation_container);
+            conversationContainer.setClipChildren(true);
+            conversationContainer.setClipToPadding(true);
+        }
+
+        setActionBarColor(recipient.getColor());
+
+        titleView.setCustomTitle(glideRequests, recipient,"Pinned Messages", "With " + recipient.getProfileName());
```

Comment:
In addition to adding recipient profile name, a backup option for numbers or chat names could be useful.